src/test/java/org/galatea/starter/entrypoint/IexRestControllerTest.java

```diff
@@ -81,4 +81,142 @@ public void testGetLastTradedPriceEmpty() throws Exception {
+    MvcResult result = this.mvc.perform(
+        org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+            .get("/iex/historicalPrices?symbol=&range=1m&date=20200409")
+            .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isNotFound())
+        .andReturn();
+  }
+
+  @Test
+  public void testHistoricalPricesDateandSymbol() throws Exception {
+    MvcResult result = this.mvc.perform(
+        org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+            .get("/iex/historicalPrices?symbol=FB&range=&date=20200409")
+            .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].high").value(new BigDecimal("175.71")))
+        .andReturn();
+  }
+
+  @Test
+  public void testHistoricalPricesSymbolOnly() throws Exception {
+    MvcResult result = this.mvc.perform(
+        org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+            .get("/iex/historicalPrices?symbol=FB&range=&date=")
+            .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].high").value(new BigDecimal("175.71")))
+        .andReturn();
+  }
+
+  @Test
+  public void testHistoricalPricesRangeOnly() throws Exception {
+    MvcResult result = this.mvc.perform(
+            org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+                .get("/iex/historicalPrices?symbol=&range=1m&date=")
+                .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isOk())
+        .andExpect(jsonPath("$[0].high").value(new BigDecimal("175.71")))
+        .andReturn();
+  }
+
+  @Test
+  public void testHistoricalPricesInvalidRange() throws Exception {
+    MvcResult result = this.mvc.perform(
+            org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+                .get("/iex/historicalPrices?symbol=FB&range=1000lodsfj&date=")
+                .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isBadRequest())
+        .andReturn();
+  }
+
+  @Test
+  public void testHistoricalPricesInvalidDate() throws Exception {
+    MvcResult result = this.mvc.perform(
+            org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+                .get("/iex/historicalPrices?symbol=FB&range=1m&date=11111111111111")
+                .accept(MediaType.APPLICATION_JSON_VALUE))
+        .andExpect(status().isBadRequest())
+        .andReturn();
+  }
+
+  @Test
+  public void testHistoricalPricesBadSymbol() throws Exception {
+    MvcResult result = this.mvc.perform(
+            org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+                .get("/iex/historicalPrices?symbol=LOKOKLKOK&range=1m&date=")
+                .accept(MediaType.APPLICATION_PROBLEM_JSON))
+        .andExpect(status().isNotAcceptable())
```

Comment:
Why is this one different than the above two tests which return BAD_REQUEST?

Reply:
We throw the BAD_REQUEST manually for the other two tests (invalid date and invalid range). But the bad symbol is handled by the Cloud API already. So I am just expecting the error I get from the Cloud API if the symbol is not found! Which happens to be Error 406 (NOT_ACCEPTABLE)